### PR TITLE
Vue amplify button

### PR DESCRIPTION
**Pull request by ErikCH**
*Issue #, if available:*
Added new Amplify Button that customers can use

*Description of changes:*

This will help with our slot documentation. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/vue/src/components/primitives/amplify-button.vue [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+interface PropsInterface {
+  type?: 'submit' | 'button';
+  fullWidth?: boolean | string;
+  size?: 'small' | 'medium' | 'large';
+  variation?: 'primary' | 'default' | 'link';
+  fontWeight?: 'normal' | 'bold' | 'lighter';
+}
+const { type, fullWidth, size, variation, fontWeight } = withDefaults(
+  defineProps<PropsInterface>(),
+  {}
```
Comment (ErikCH) [inline packages/vue/src/components/primitives/amplify-button.vue:11]: I left off defaults now, will update at a future time, if needed

File: packages/vue/src/components/primitives/amplify-button.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+);
+</script>
+<template>
+  <slot v-bind="$attrs" name="buttont">
+    <button
+      :type="type"
```
Comment (wlee221) [inline packages/vue/src/components/primitives/amplify-button.vue:17]: Can we add 
```suggestion
    <button
      class="amplify-button"
      :type="type"
```
so these are styled by default?

Comment (ErikCH) [reply packages/vue/src/components/primitives/amplify-button.vue]: Sure, just pushed

**Review by wlee221 — APPROVED** [review]
Thanks! LGTM.

**Pull request merged**